hela/forest.py

```diff
@@ -0,0 +1,251 @@
+    print("Testing scores:")
+    for name, values in r2scores.items():
+        print("\tR^2 score for {}: {:.3f}".format(name, values))
+
+    fig = plot_predicted_vs_real(dataset.testing_y, pred, dataset.names,
+                                 dataset.ranges)
+    fig.savefig(os.path.join(output_path, "predicted_vs_real.pdf"),
+                bbox_inches='tight')
+    return r2scores
+
+
+def compute_feature_importance(model, dataset, output_path):
+    regr = multioutput.MultiOutputRegressor(model, n_jobs=1)
+    regr.fit(dataset.training_x, dataset.training_y)
+
+    forests = [i.rf for i in regr.estimators_] + [model.rf]
+
+    fig = plot_feature_importances(
+        forests=[i.rf for i in regr.estimators_] + [model.rf],
+        names=dataset.names + ["joint prediction"],
+        colors=dataset.colors + ["C0"])
+
+    fig.savefig(os.path.join(output_path, "feature_importances.pdf"),
+                bbox_inches='tight')
+    return np.array([forest_i.feature_importances_ for forest_i in forests])
+
+
+class RandomForest(object):
+    """
+    A class for a random forest.
+    """
+
+    def __init__(self, training_dataset, model_path, data_file):
+        """
+        Parameters
+        ----------
+        training_dataset : str
+            Path to the dataset metadata JSON file
+        model_path : str
+            Path to the output directory to create and populate
+        data_file : str
+            Path to the numpy pickle of the samples to predict on
+        """
+        self.training_dataset = training_dataset
+        self.model_path = model_path
+        self.data_file = data_file
+        self.output_path = self.model_path
+
+        self.dataset = None
+        self.model = None
+
+    def train(self, num_trees=1000, num_jobs=5, quiet=False):
+        """
+        Train the random forest on a set of observations.
+
+        Parameters
+        ----------
+        num_trees : int
+        num_jobs : int
+        quiet : bool
+
+        Returns
+        -------
+        r2scores : dict
+            :math:`R^2` values for each parameter after training
+        """
+        # Loading dataset
+        self.dataset = load_dataset(self.training_dataset)
+
+        # Training model
+        self.model = train_model(self.dataset, num_trees, num_jobs, not quiet)
+
+        os.makedirs(self.model_path, exist_ok=True)
+        model_file = os.path.join(self.model_path, "model.pkl")
+        # Saving model
+        joblib.dump(self.model, model_file)
+
+        # Printing model information...
+        print("OOB score: {:.4f}".format(self.model.rf.oob_score_))
+
+        r2scores = test_model(self.model, self.dataset, self.model_path)
+
+        return r2scores
+
+    def feature_importance(self):
+        """
+        Compute feature importance.
+
+        Returns
+        -------
+        feature_importances : `~numpy.ndarray`
+        """
+        self.model.enable_posterior = False
```

Comment:
This line is wrong now. I will fix this.

Reply:
I've seen that you have removed this line, but the problem is more intricate. If you just remove the line without changing the implementation of `RandomForestWrapper`, the feature computation will take MUCH more time. I think I will change the way feature importance is computed in a subsequent PR.